provider/src/main/java/com/cloudera/director/google/sql/GoogleCloudSQLProvider.java

```diff
@@ -0,0 +1,608 @@
+  }
+
+  @Override
+  public void allocate(GoogleCloudSQLInstanceTemplate template,
+      Collection<String> instanceIds, int minCount) throws InterruptedException {
+
+    PluginExceptionConditionAccumulator accumulator = new PluginExceptionConditionAccumulator();
+
+    LocalizationContext providerLocalizationContext = getLocalizationContext();
+    LocalizationContext templateLocalizationContext =
+        SimpleResourceTemplate.getTemplateLocalizationContext(providerLocalizationContext);
+
+    for (String instanceId : instanceIds) {
+      String decoratedInstanceName = decorateInstanceName(template, instanceId, templateLocalizationContext);
+      if (!instanceNamePrefixPattern.matcher(decoratedInstanceName).matches()) {
+        throw new InterruptedException("The database instance name " + decoratedInstanceName + " doesn't meet Google Cloud SQL requirements. It can only contain lowercase letters, numbers and hyphens and start with a lowercase letter.");
+      }
+    }
+
+    SQLAdmin sqlAdmin = credentials.getSQLAdmin();
+    String projectId = credentials.getProjectId();
+
+    // Use this list to collect the operations that must reach a RUNNING or DONE state prior to allocate() returning.
+    List<Operation> dbCreationOperations = new ArrayList<Operation>();
+    List<Operation> userCreationOperations = new ArrayList<Operation>();
+    List<String> successfullyCreatedInstancesNames = new ArrayList<String>();
+
+    for (String instanceId : instanceIds) {
+      String decoratedInstanceName = decorateInstanceName(template, instanceId, templateLocalizationContext);
+
+      // Compose the settings.
+      Settings settings = new Settings();
+
+      String tierName = template.getConfigurationValue(TIER, templateLocalizationContext);
+      settings.setTier(tierName);
+
+      IpConfiguration ipConfiguration = new IpConfiguration();
+      ipConfiguration.setIpv4Enabled(true);
+
+      // TODO(kl3n1nz) Might want to whitelist only GCE instances.
+      AclEntry aclEntry = new AclEntry();
+      aclEntry.setValue("0.0.0.0/0");
+      aclEntry.setKind("sql#aclEntry");
+      aclEntry.setName("world");
+
+      ipConfiguration.setAuthorizedNetworks(Arrays.asList(aclEntry));
+      settings.setIpConfiguration(ipConfiguration);
+
+      String preferredLocation = template.getConfigurationValue(PREFERRED_LOCATION, templateLocalizationContext);
+      if (preferredLocation != null && !preferredLocation.isEmpty()) {
+
+        // Set the preferred location.
+        LocationPreference locationPreference = new LocationPreference();
+        locationPreference.setKind("sql#locationPreference");
+        locationPreference.setZone(preferredLocation);
+        settings.setLocationPreference(locationPreference);
+      }
+
+      // Compose the instance.
+      DatabaseInstance instance = new DatabaseInstance();
+      String regionName = template.getConfigurationValue(REGION_SQL, templateLocalizationContext);
```

Comment:
No, because I list the correct value in the property. It becomes an issue if I try to match GCE regions and zones with CloudSQL regions.